rpcserver.go

```diff
@@ -4521,16 +4521,55 @@ func (r *rpcServer) FeeReport(ctx context.Context,
 // 0.000001, or 0.0001%.
 const minFeeRate = 1e-6
 
+// checkMaxHtlc errors if the user tries to update their channel policy with
+// a max HTLC value that is too high or too low for a given list of channels.
+func checkMaxHtlc(maxHtlc uint64, channels []*channeldb.OpenChannel) error {
```

Comment:
Sort of, the `rpcserver` checks that a MaxHtlc is below the `MaxPendingAmount` for our channel, but the `gossiper` only checks that it's below the channel capacity, since it doesn't have access to `MaxPendingAmount` info (iinm). 

Still, lmk if you think this check should just be removed -- it creates a bit of inconsistency, even though `MaxPendingAmount` is a better upper limit.

Reply:
I think that ideally you want to have the check only in one place and the `ValidateChannelUpdateAnn` is already there. Gossiper is also already querying `Router` for outgoing channels. What would it take to also retrieve the `MaxPendingAmount` there and check it?

Reply:
Some options:
1. In the gossiper's `updateChannel`, use the `ChannelEdgePolicy`'s `DB` pointer to get the `MaxPendingAmount`
2. Add a closure for fetching the `MaxPendingAmount` of a channel to the gossiper's `cfg`
3. Pass [`targetDBChans`](https://github.com/lightningnetwork/lnd/pull/2438/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR4560) to the gossiper's `PropagateChanPolicyUpdate`

Reply:
I'd think that `Router.ForAllOutgoingChannels` is the place to do it. Return more complete info on the set of outgoing channels that includes `MaxPendingAmount`.

Reply:
Discussed with @halseth what a final design could be to move towards in small steps. It seems that having a distinct object `LocalChannelManager` that manages our local channel state including the forwarding policy would be an improvement. It could expose `ForAllOutgoingChannels`, but also something like `ValidatePolicy`. This also allows to, later, split all the persistent data associated with our own channels to a separate db. When an updates happens in `LocalChannelManager`, it is pushed to the gossiper. No further checking is needed in the gossiper anymore.

I think that ideally we'd want to make a step in this PR towards that. Or towards another final design that we think is good. The current change with another loop over channels in `rpcserver` feels to me like creating technical debt. For an urgent bug fix I can understand that, but if we are not in a hurry with this max htlc setting, I'd prefer doing it right.